nbs/04_processing.ipynb

```diff
@@ -203,61 +205,74 @@
-    "                                                                                                   grid_configs = self.grid_configs)\n",
-    "            label_id_overview_fig.savefig(self.results_dir_path.joinpath('ROI_label_IDs_overview.png'), dpi = 300)\n",
+    "            self.activity_overview_plot[0].savefig(self.results_dir_path.joinpath('activity_overview.png'), dpi=300)\n",
+    "            label_id_overview_fig, label_id_overview_ax = results.plot_rois_with_label_id_overview(\n",
+    "                analysis_rois_with_sufficient_activity=activity_filtered_analysis_rois,\n",
+    "                preview_image=self.recording.preview,\n",
+    "                focus_area=self.focus_area,\n",
+    "                grid_configs=self.grid_configs)\n",
+    "            label_id_overview_fig.savefig(self.results_dir_path.joinpath('ROI_label_IDs_overview.png'), dpi=300)\n",
     "            plt.close()\n",
     "        if config.save_summary_results:\n",
     "            self._create_and_save_csv_result_files(activity_filtered_analysis_rois)\n",
     "            self._create_and_save_individual_traces_pdf_result_file(activity_filtered_analysis_rois)\n",
     "        if config.save_single_trace_results:\n",
     "            self._create_and_save_single_trace_results_as_csv(activity_filtered_analysis_rois)\n",
     "        if config.export_to_nwb:\n",
-    "            self._export_all_data_to_nwb()\n",
+    "            self._export_all_data_to_nwb(nwb_metadata)\n",
     " \n",
-    "    def _export_all_data_to_nwb(self) -> None:\n",
-    "        nwb_file = self._initialize_nwb_file()\n",
-    "        nwb_file = self._add_source_data(nwb_file)\n",
+    "    def _export_all_data_to_nwb(self, nwb_metadata: dict[str, Any] = None) -> None:\n",
+    "        if nwb_metadata is None:\n",
+    "            nwb_metadata = {}\n",
+    "        nwb_file = self._initialize_nwb_file(nwb_metadata)\n",
+    "        nwb_file = self._add_source_data(nwb_file, nwb_metadata)\n",
     "        nwb_file = self._add_na3_results(nwb_file)\n",
     "        self._write_nwb_file(nwb_file)\n",
     "\n",
     "\n",
-    "    def _initialize_nwb_file(self) -> NWBFile:\n",
+    "    def _initialize_nwb_file(self, nwb_metadata: dict[str, Any]) -> NWBFile:\n",
     "        nwb_file = NWBFile(\n",
-    "            session_description=\"Autogenerated NWB file by neuralactivitycubic\",\n",
+    "            session_description=\"Autogenerated NWB file by Neuralactivitycubic\",\n",
     "            identifier=str(uuid4()),\n",
-    "            session_start_time=datetime.now(tzlocal())\n",
+    "            session_start_time=datetime.now(tzlocal()),\n",
+    "            experimenter=nwb_metadata.get('experimenter', 'NA3 user'),\n",
+    "            experiment_description=nwb_metadata.get('experiment_description', 'Experiment done with NA3'),\n",
```

Comment:
This should be left blank, considering that the experiment - i.e. data acquisition - was not done with NA3, which was only used for data analysis.